### PR TITLE
refactor widget controls

### LINKED ISSUE #1720: Bugfix: All country dropdowns appear in the first chart widget in the report.
## Expected Behavior
Each chart widget receives its own country dropdown.

## Current Behavior
All country dropdowns appear in the first chart widget.

## Possible Solution
more specific selectors

## Steps to Reproduce

1.
2.
3.
4.

### LINKED ISSUE #1721: Feature: Refactor widget controls.
## Feature Details
improve structure of functions that add controls to widgets

### PR DESCRIPTION
**Pull request by samussiah**
## Overview
- Resolves #1720 
- Resolves #1721 

## Test Notes/Sample Code
Run `inst/examples/3_ReportingWorkflow.R` to generate a report with the updates.

Notes:

### CODE REVIEW

File: inst/utils-0.0.1/addWidgetControls.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+ *
+ * @param {Node} el - widget container, an element in the DOM
+ * @param {Array} dfResults - analysis results for a single metric
+ * @param {Object} lMetrics - metric metadata
+ * @param {Array} dfGroups - group metadata
+ *
+ * @returns {Node} HTML select element
+ */
+const addWidgetControls = function(el, dfResults, lMetrics, dfGroups, bAddGroupSelect) {
+    if (bAddGroupSelect === false)
```
Comment (jonthegeek) [inline inst/utils-0.0.1/addWidgetControls.js:12]:
```suggestion
    if (!bAddGroupSelect)
```
I *think* we want to return the version without a control when bAddGroupSelect is malformed, too (not just when it's exactly `false`).

**Review by jonthegeek — CHANGES_REQUESTED** [review]
One tweak and a question. Looking really good overall!

File: inst/utils-0.0.1/addClickCallback.js
[the remark this answers is not in the document]
Comment (samussiah) [reply inst/utils-0.0.1/addClickCallback.js:1]: You got it - I wanted to pull the click callbacks into a dedicated function but it ended up being a little more involved than I thought.

**Review by jonthegeek — APPROVED** [review]
Looks good!

**Pull request merged**